roles/ceph-common/tasks/installs/install_rh_storage_on_redhat.yml

```diff
@@ -0,0 +1,158 @@
+---
+- name: install red hat storage repository key
+  rpm_key:
+    key: "{{ ceph_stable_rh_storage_repository_path }}/RPM-GPG-KEY-redhat-release"
+    state: present
+  when:
+    - ceph_stable_rh_storage
+    - ceph_stable_rh_storage_iso_install
+
+- name: add red hat storage repository
+  template:
+    src: ../../templates/redhat_storage_repo.j2
+    dest: /etc/yum.repos.d/rh_storage.repo
+    owner: root
+    group: root
+    mode: 0644
+  when:
+    - ceph_stable_rh_storage
+    - ceph_stable_rh_storage_iso_install
+
+- name: install dependencies
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: redhat_package_dependencies
+  when:
+    - ansible_distribution == "RedHat"
+    - ansible_pkg_mgr == "yum"
+
+- name: install dependencies
+  yum:
+    name: "{{ item }}"
+    state: present
+  with_items: centos_package_dependencies
+  when:
+    - ansible_distribution == "CentOS"
+    - ansible_pkg_mgr == "yum"
+
+- name: install dependencies
+  dnf:
+    name: "{{ item }}"
+    state: present
+  with_items: centos_package_dependencies
+  when:
+    - ansible_distribution == "CentOS"
+    - ansible_pkg_mgr == "dnf"
+
+- name: install red hat storage ceph mon (yum)
+  yum:
+    name: "ceph-mon"
+    state: "{{ (upgrade_ceph_packages|bool) | ternary('latest','present') }}"
+  when:
+    - mon_group_name in group_names
+    - ansible_pkg_mgr == "yum"
+    - (ceph_stable and ceph_stable_release not in ceph_stable_releases)
+      or ceph_stable_rh_storage
+
+- name: install red hat storage ceph mon (dnf)
+  dnf:
+    name: "ceph-mon"
+    state: "{{ (upgrade_ceph_packages|bool) | ternary('latest','present') }}"
+  when:
+    - mon_group_name in group_names
+    - ansible_pkg_mgr == "dnf"
+    - (ceph_stable and ceph_stable_release not in ceph_stable_releases)
```

Comment:
I never quite understood the intention of ``(ceph_stable and ceph_stable_release not in ceph_stable_releases)``, but I don't think you need it here anymore.